zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchStorage.java

```diff
@@ -296,21 +294,21 @@ String buildUrl(IndexTemplates templates, String type) {
     return new HttpCall.Factory(lazyHttpClient().get());
   }
 
-  @Override public void close() throws IOException {
+  @Override public void close() {
     lazyHttpClient().close();
   }
 
   ElasticsearchStorage() {
   }
 
   static final BodyConverter<CheckResult> READ_STATUS = new BodyConverter<CheckResult>() {
-    @Override public CheckResult convert(HttpData body) throws IOException {
-      // The health check is only invoked periodically. Hence, there is less impact to allocating
-      // strings. We retain the string so that it can be logged if the ES response is malformed.
-      String result = body.toStringUtf8();
-      JsonParser status = enterPath(JSON_FACTORY.createParser(result), "status");
+    @Override public CheckResult convert(JsonParser parser, Supplier<String> contentString)
+      throws IOException {
+      JsonParser status = enterPath(parser, "status");
       if (status == null) {
-        throw new IllegalArgumentException("Health status couldn't be read " + result);
+        // The health check is only invoked periodically. Hence, there is less impact to allocating
```

Comment:
Don't think you need this comment anymore.

Reply:
agreed